### ISSUE TITLE
Document =pod comment expectation dangers

### ISSUE THREAD

**Issue by jidanni** [briandfoy/pod-perldoc#60]
See https://github.com/Perl/perl5/issues/21688#issuecomment-1846519022 .

**Comment by briandfoy** [comment]
Can you condense what you want in this thread and what you think we should do about it? It would help especially in this case because I read your comments there as asking two opposite things.

**Comment by briandfoy** [comment]
I know that `perldoc` is a frustrating bit of code, but it seems like something that is simple when it actually is not. When we consider what to document, we have to examine whether what we write is true (or true enough), and in the right place. Generally, for something like `perldoc`, any behavior that we want to document has to generally apply. I don't think that's true for the change you suggest. I've thought a lot about how we might document this particular issue, and there's not a simple couple of sentences that we could put somewhere that wouldn't raise all sorts of other issues that we'd need to document.

perldoc doesn't work the way that you seem to think it does. `perldoc` is a dispatcher to a Pod formatter and is not actually doing the work to translate Pod. These formatters aren't even part of the Pod::Perldoc distribution. Pod::Perldoc doesn't even know ahead of time which formatter might be used for a particular output format (see`init_formatter_class_list`, as I mentioned to you in #59). `perldoc` asks some formatter to take some input, then displays whatever that formatter produces. It does not tell the formatter how to parse the input, what to include, what to exclude, or how to think about what it encounters. Consider `=begin`, `=end`, and `=for`. Not every translator handles every use of those. Compare and contrast how Pod::Text and Pod::HTML work, for instance. Or, see translators such as my Pod::Perldoc::ToToc.

A good analogy which be that you want to document something about how a train car works, but we're just the switching yard. We only care that the train car follows the tracks, and beyond that it can be whatever it needs to be. I think the best we could do is document how `perldoc` decides if a source has Pod and when it decides not to send input to a translator (see `containspod`). This is a whole other tricky situation that we have to guess at since perldoc does not actually parse the input (again, that's the translator's work). 

Here's another analogy: you are asking your math textbook to explain your calculator. Your math book doesn't know what calculator you have. Maybe you think that most calculators are the same, but then you experience an RPN calculator (oh how I mourn the death of my HP-48) and realize it's completely different. 

Although most Pod translators that Pod::Perldoc might use in most cases act similarly, it's up to the particular Pod translator to decide what to do with its. As such, we cannot document in `perldoc` itself that every Pod translator will act the same (or not act in the same way). The `perldoc` modules dispatch work to modules outside of the Pod::Perldoc repo, such as Pod::Man and Pod::Text. Those external modules decide what to do, not `perldoc`. 

Along with that, people can use translators that we don't even know about by specifying part of its name with `-o`. `perldoc` then looks for module names that match that in various ways. It might find several candidate modules, but choose different candidates in different situations.

As such, we can't make one statement about any sort of behavior that `perldoc` does not control or enforce. We can say that the behavior is up to the individual formatters, but I think that's generally understood. The right place for this sort of fix is in the translator itself since `perldoc` doesn't know what those translators are going to do. 

To summarize:

* We can't document this in perldoc because perldoc doesn't make that decision
* Different translators make different decisions
* We don't know all of the translators involved and what they might do
* None of these translators are in this distribution
